pkg/sdk/database_role_test.go

```diff
@@ -0,0 +1,243 @@
+		}
+	}
+
+	t.Run("validation: nil options", func(t *testing.T) {
+		var opts *alterDatabaseRoleOptions = nil
+		assertOptsInvalidJoinedErrors(t, opts, errNilOptions)
+	})
+
+	t.Run("validation: incorrect identifier", func(t *testing.T) {
+		opts := setUpOpts()
+		opts.name = NewDatabaseObjectIdentifier("", "")
+		assertOptsInvalidJoinedErrors(t, opts, ErrInvalidObjectIdentifier)
+	})
+
+	t.Run("validation: no alter action", func(t *testing.T) {
+		opts := setUpOpts()
+		assertOptsInvalidJoinedErrors(t, opts, errAlterNeedsExactlyOneAction)
+	})
+
+	t.Run("validation: multiple alter actions", func(t *testing.T) {
+		opts := setUpOpts()
+		opts.Set = &DatabaseRoleSet{
+			Comment: "new comment",
+		}
+		opts.Unset = &DatabaseRoleUnset{
+			Comment: true,
+		}
+		assertOptsInvalidJoinedErrors(t, opts, errAlterNeedsExactlyOneAction)
+	})
+
+	t.Run("validation: invalid new name", func(t *testing.T) {
+		opts := setUpOpts()
+		opts.Rename = &DatabaseRoleRename{
+			Name: NewDatabaseObjectIdentifier("", ""),
+		}
+		assertOptsInvalidJoinedErrors(t, opts, ErrInvalidObjectIdentifier)
+	})
+
+	t.Run("validation: new name from different db", func(t *testing.T) {
+		newId := NewDatabaseObjectIdentifier(id.DatabaseName()+randomStringN(t, 1), randomStringN(t, 12))
+
+		opts := setUpOpts()
+		opts.Rename = &DatabaseRoleRename{
+			Name: newId,
+		}
+		assertOptsInvalidJoinedErrors(t, opts, errDifferentDatabase)
+	})
+
+	t.Run("validation: no property to unset", func(t *testing.T) {
+		opts := setUpOpts()
+		opts.Unset = &DatabaseRoleUnset{
+			Comment: false,
+		}
+		assertOptsInvalidJoinedErrors(t, opts, errAlterNeedsAtLeastOneProperty)
+	})
+
+	t.Run("rename", func(t *testing.T) {
+		newId := NewDatabaseObjectIdentifier(id.DatabaseName(), randomStringN(t, 12))
+
+		opts := setUpOpts()
+		opts.Rename = &DatabaseRoleRename{
+			Name: newId,
+		}
+		assertOptsValidAndSQLEquals(t, opts, `ALTER DATABASE ROLE %s RENAME TO %s`, id.FullyQualifiedName(), newId.FullyQualifiedName())
+	})
+
+	t.Run("set", func(t *testing.T) {
+		opts := setUpOpts()
+		opts.IfExists = Bool(true)
+		opts.Set = &DatabaseRoleSet{
+			Comment: "new comment",
+		}
+		assertOptsValidAndSQLEquals(t, opts, `ALTER DATABASE ROLE IF EXISTS %s SET COMMENT = 'new comment'`, id.FullyQualifiedName())
+	})
+
+	t.Run("set comment to empty", func(t *testing.T) {
+		opts := setUpOpts()
+		opts.IfExists = Bool(true)
+		opts.Set = &DatabaseRoleSet{
+			Comment: "",
+		}
+		assertOptsValidAndSQLEquals(t, opts, `ALTER DATABASE ROLE IF EXISTS %s SET COMMENT = ''`, id.FullyQualifiedName())
+	})
+
+	t.Run("unset", func(t *testing.T) {
+		opts := setUpOpts()
+		opts.IfExists = Bool(true)
+		opts.Unset = &DatabaseRoleUnset{
+			Comment: true,
+		}
+		assertOptsValidAndSQLEquals(t, opts, `ALTER DATABASE ROLE IF EXISTS %s UNSET COMMENT`, id.FullyQualifiedName())
+	})
+}
+
+func TestDatabaseRoleDrop(t *testing.T) {
+	id := randomDatabaseObjectIdentifier(t)
+
+	setUpOpts := func() *dropDatabaseRoleOptions {
```

Comment:
maybe this should be called `defaultOpts()`? and then perhaps do something like:

```
defaultOpts := setupOpts[T](opts T) func()T
e.g.
defaultOpts := setupOpts(&dropDatabaseRoleOptions{name:id})

```

so then later you can do:
```
opts  := defaultOpts()
opts.IfExists = Bool(true)
```
this

Reply:
I can rename it, but for the second comment, we are already using it like this, so I don't think that change is needed here.

Reply:
Rename done.